### PR TITLE
Add PostgreSql connection string prefix

### LINKED ISSUE #47704: Support POSTGRESQLCONNSTR_ in EnvironmentVariablesConfigurationProvider.cs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I am trying to create a Azure App Service with Postgres DB (Web app + Database option on the portal)
The connection string I've been provided with has a prefix POSTGRESQLCONNSTR_ in the environment, which is not recognized by EnvironmentVariablesConfigurationProvider.cs, so I have to either access it as environment variable directly or to add the prefix as a parameter to the .AddEnvironmentVariables(). The MSSQL, SQLAzure, MySQL connection string environment variables are processed properly.

### Describe the solution you'd like

I would like POSTGRESQLCONNSTR_ to be added as recognizable connection string prefix in EnvironmentVariablesConfigurationProvider.cs.

### Additional context

conn_str
conn_str as environment variable

### PR DESCRIPTION
**Pull request by bookuha**
closes [#47704](https://github.com/dotnet/aspnetcore/issues/47704)

### CODE REVIEW

**Comment by bookuha** [comment]
@dotnet-policy-service agree

**Comment by bookuha** [comment]
> @bookuha please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

**Comment by bookuha** [comment]
@dotnet-policy-service agree

File: src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class EnvironmentVariablesConfigurationProvider : ConfigurationProvider
         private const string MySqlServerPrefix = "MYSQLCONNSTR_";
         private const string SqlAzureServerPrefix = "SQLAZURECONNSTR_";
         private const string SqlServerPrefix = "SQLCONNSTR_";
+        private const string PostrgeSqlPrefix = "POSTGRESQLCONNSTR_"
```
Comment (kant2002) [inline src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs:18]: I think you should use this in location around this
https://github.com/dotnet/runtime/blob/3d97f41f084953ff7dc85776b82f976e6a22919e/src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs#L67-L70

Comment (kant2002) [reply src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs:18]: Also you should add test for this here https://github.com/dotnet/runtime/blob/3d97f41f084953ff7dc85776b82f976e6a22919e/src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/tests/EnvironmentVariablesTest.cs#L64-L65

Comment (bookuha) [reply src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs:18]: My bad. Adding the missing part

**Comment by tarekgh** [comment]
@bookuha please have a look at https://github.com/dotnet/runtime/issues/36123. I am going to close this PR till this issue is handled. 

@davidfowl please comment on that.

**Pull request closed without merging**